docs/content/TextInput.mdx

```diff
@@ -108,6 +108,58 @@ const WithIconAndLoadingIndicator = () => {
 render(<WithIconAndLoadingIndicator />)
 ```
 
+### With trailing action
+
+```jsx live
+<Box display="grid" sx={{gap: 3}}>
+  <FormControl>
+    <FormControl.Label>Icon action</FormControl.Label>
+    <TextInput
+      trailingAction={
+        <TextInput.Action
+          onClick={() => {
+            alert('clear input')
+          }}
+          icon={XIcon}
+          iconLabel="Clear input"
+          sx={{color: 'fg.subtle'}}
+        />
+      }
+    />
+  </FormControl>
+  <FormControl>
+    <FormControl.Label>Icon action with tooltip</FormControl.Label>
+    <TextInput
+      trailingAction={
+        <TextInput.Action
+          onClick={() => {
+            alert('clear input')
+          }}
+          icon={XIcon}
+          iconLabel="Clear input"
+          tooltipMessage="Clear"
+          sx={{color: 'fg.subtle'}}
+        />
```

Comment:
Is there a reason we couldn't wrap TextInput.Action with a Tooltip?

```jsx
<Tooltip aria-label="Clear"><TextInput.Action ... /></Tooltip>
```

Reply:
I imagine consumers don't to want to think about when they should use a tooltip

Reply:
I vote we replace `iconLable` and `tooltipMessage` with an `aria-label` prop that is required if no children are passed

Reply:
@siddharthkp - I thought there would be a case where the `iconLabel` and `tooltipMessage` are different, but now I'm second-guessing that... 🤔 

I'm going to move forward with @colebemis's suggestion, and we can change that later if we need to.

Reply:
Should we start without allowing a tooltip for buttons with text labels? This feels like a reasonable limitation, but I don't like that we would just silently swallow the `aria-label`

Reply:
> Should we start without allowing a tooltip for buttons with text labels? This feels like a reasonable limitation

That sounds reasonable!

> , but I don't like that we would just silently swallow the aria-label

For actions with text labels, is there a reason not to pass aria-label through to the Button if it's provided?

Reply:
I decided not to limit it